File: aws/cdk/java/src/main/java/com/gatlingenterprise/PrivateLocation.java
package com.gatlingenterprise;

import com.gatlingenterprise.records.controlPlaneProps.ControlPlaneProps;
import com.gatlingenterprise.records.locationProps.AMI;
import com.gatlingenterprise.records.locationProps.EnterpriseCloud;
import com.gatlingenterprise.records.locationProps.Location;
import com.gatlingenterprise.records.locationProps.TagsFor;
import java.util.List;
import java.util.Map;
import software.amazon.awscdk.App;

public class PrivateLocation {
  public static void main(final String[] args) {
    App app = new App();

    String stackName = "GatlingEnterprise-PrivateLocation-Stack";

    Location location =
        new Location(
            "prl_aws", // id
            "Private Location on AWS", // desc
            "eu-west-1", // region
            List.of("subnet-1"), // subnets
            List.of("sg-id"), // securityGroups
            "c7i.xlarge", // instanceType
            false, // spot
            new AMI("certified", "latest", null), // ami
            "classic", // engine
            List.of(), // elasticIps
            Map.of(), // tags
            new TagsFor(null, null, null), // tagsFor (set to null if absent)
            "", // profileName
            "", // iamInstanceProfile
            Map.of(), // systemProperties
            "/usr/lib/jvm/zulu", // javaHome
            List.of(), // jvmOptions
            new EnterpriseCloud("https://api.gatling.io") // Location forward proxy url
            );

    new ControlPlaneStack(
        app,
        stackName,
        new ControlPlaneProps(
            "vpc-id", // vpcId
            List.of("eu-west-1a", "eu-west-1b", "eu-west-1c"), // availabilityZones
            List.of("subnet-1", "subnet-2", "subnet-3"), // subnetIds
            List.of("sg-id"), // securityGroupIds
            "token", // token
            "GatlingEnterprise/ControlPlane/Token", // tokenSecretName
            "gatling-pl", // name
            "My AWS control plane description", // desc
            "gatlingcorp/control-plane:latest", // image
            List.of(), // command
            Map.of(), // environment
            Map.of(), // secrets
            List.of(location), // locations
            null, // privatePackage
            true, // cloudWatchLogs
            false, // useECR
            new EnterpriseCloud("https://api.gatling.io") // Control plane forward proxy url
            ));

    app.synth();
  }
}

File: aws/cdk/typescript/bin/private-location.ts
#!/usr/bin/env node
import "source-map-support/register";
import { App } from "aws-cdk-lib";
import { ControlPlaneStack } from "../lib/control-plane";

const app = new App();
const stackName = "GatlingEnterprise-PrivateLocation-Stack";

const location = {
  // Possible values available on interface Location in lib/interfaces/common-interface.ts
  id: "prl_aws",
  description: "Private Location on AWS",
  region: "‹Region>",
  subnets: ["<SubnetId>"],
  "security-groups": ["<SecurityGroupId>"],
  "instance-type": "c7i.xlarge",
  // engine: "classic",
  ami: {
    type: "certified",
    //   java: "latest",
    //   id: "ami-00000000000000000",
  },
  // engine: "classic",
  // spot: false,
  // "auto-associate-public-ipv4": true,
  // "elastic-ips": ["203.0.113.3"],
  // "profile-name": "profile-name",
  // "iam-instance-profile": "iam-instance-profile",
  // tags: {},
  // "tags-for": {
  //   instance: {},
  //   volume: {},
  //   "network-interface": {},
  // },
  // "system-properties": {},
  // "java-home": "/usr/lib/jvm/zulu",
  // "jvm-options": [],
  // "enterprise-cloud": {
  //   // Setup the proxy configuration for the private location
  //   // Reference: https://docs.gatling.io/reference/install/cloud/private-locations/network/#configuring-a-proxy
  // },
};

new ControlPlaneStack(app, stackName, {
  // Possible values available on interface ControlPlaneProps in lib/interfaces/control-plane-interface.ts
  name: "<Name>",
  description: "My AWS control plane description",
  tokenSecretArn: "<TokenSecretARN>",
  vpcId: "<VPCId>",
  availabilityZones: ["<AZ>"],
  subnets: ["<SubnetId>"],
  securityGroups: ["<SecurityGroupId>"],
  locations: [location],
  // task: {
  //   cpu: 1024,
  //   memory: 3072,
  //   init: {
  //     image: "busybox",
  //   },
  //   image: "gatlingcorp/control-plane:latest",
  //   command: [],
  //   secrets: {},
  //   environment: {},
  //   cloudwatchLogs: true,
  //   ecr: false,
  // },
  // enterpriseCloud: {
  //   // Setup the proxy configuration for the control plane
  //   // Reference: https://docs.gatling.io/reference/install/cloud/private-locations/network/#configuring-a-proxy
  // },
});
